h2o-core/src/main/java/water/util/FileUtils.java

```diff
@@ -31,6 +31,7 @@ public static void copyStream(InputStream is, OutputStream os, final int buffer_
           break;
         os.write(bytes, 0, count);
       }
+      os.close();
```

Comment:
Note to self: Investigate this line. Why should we close a stream that was given to the method ? Is that really part of method's contract ?

Reply:
I think that it should be here, since the entire stream is read and without this line it's not flushed to disk.

Reply:
hmm, in my opinion, it should really be a responsibility of the caller. Only the caller knows that it will not use eg. mark() and reset() methods on the stream and attempt to rewind it back.

`IOUtils.copy` is not closing the input stream. We should take it as a reference. 

Problem is there are incorrect uses of the copyStream in our code bases and streams are not closed. That should be fixed

Reply:
Fixed in 838b5b8.